core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -216,6 +221,8 @@ private[spark] object ReliableCheckpointRDD extends Logging {
         serializeStream.writeObject(partitioner)
       } {
         serializeStream.close()
+        fileOutputStream.close()
```

Comment:
Ditto, this is OK if `serializeStream.close()` doesn't actually close the underlying stream (?) but not sure about the next line.